core/crdt/stats.go

```diff
@@ -0,0 +1,306 @@
+	ds "github.com/ipfs/go-datastore"
+	dsquery "github.com/ipfs/go-datastore/query"
+	blockstore "github.com/ipfs/boxo/blockstore"
+	blockservice "github.com/ipfs/boxo/blockservice"
+	syncds "github.com/ipfs/go-datastore/sync"
+	offline "github.com/ipfs/boxo/exchange/offline"
+	dagservice "github.com/ipfs/boxo/ipld/merkledag"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// StatsTopicPrefix is the pubsub topic prefix for CRDT stats synchronization
+	StatsTopicPrefix = "/warpnet/stats/crdt/1.0.0"
+)
+
+// TweetStats represents aggregated statistics for a tweet using CRDT
+type TweetStats struct {
+	TweetID       string
+	LikesCount    uint64
+	RetweetsCount uint64
+	RepliesCount  uint64
+	ViewsCount    uint64
+}
+
+// StatType represents the type of statistics
+type StatType string
+
+const (
+	StatTypeLikes    StatType = "likes"
+	StatTypeRetweets StatType = "retweets"
+	StatTypeReplies  StatType = "replies"
+	StatTypeViews    StatType = "views"
+)
+
+// CRDTStatsStore manages CRDT-based tweet statistics
+type CRDTStatsStore struct {
+	crdt        *crdt.Datastore
+	pubsub      *pubsub.PubSub
+	broadcaster *crdt.PubSubBroadcaster
+	topic       *pubsub.Topic
+	sub         *pubsub.Subscription
+	ctx         context.Context
+	cancel      context.CancelFunc
+	mu          sync.RWMutex
+	nodeID      string
+	namespace   string
+}
+
+// NewCRDTStatsStore creates a new CRDT-based statistics store
+func NewCRDTStatsStore(
+	ctx context.Context,
+	baseStore ds.Batching,
+	ps *pubsub.PubSub,
+	nodeID string,
+	namespace string,
+) (*CRDTStatsStore, error) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	// Create a blockstore for IPLD storage
+	bstore := blockstore.NewBlockstore(syncds.MutexWrap(baseStore))
+	
+	// Create block service with offline exchange (no network fetching)
+	bsrv := blockservice.New(bstore, offline.Exchange(bstore))
+	
+	// Create a DAGService for IPLD operations
+	dagSyncer := dagservice.NewDAGService(bsrv)
+
+	// Create PubSub broadcaster
+	broadcaster, err := crdt.NewPubSubBroadcaster(ctx, ps, StatsTopicPrefix)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to create broadcaster: %w", err)
+	}
+
+	// Create CRDT datastore
+	opts := crdt.DefaultOptions()
+	opts.Logger = log.StandardLogger()
+	opts.RebroadcastInterval = 0 // Disable automatic rebroadcast
+
+	crdtStore, err := crdt.New(
+		baseStore,
+		ds.NewKey(namespace),
+		dagSyncer,
+		broadcaster,
+		opts,
+	)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to create CRDT store: %w", err)
+	}
+
+	store := &CRDTStatsStore{
+		crdt:        crdtStore,
+		pubsub:      ps,
+		broadcaster: broadcaster,
+		ctx:         ctx,
+		cancel:      cancel,
+		nodeID:      nodeID,
+		namespace:   namespace,
+	}
+
+	return store, nil
+}
+
+// IncrementStat increments a specific statistic for a tweet
+func (s *CRDTStatsStore) IncrementStat(tweetID string, statType StatType) (uint64, error) {
```

Comment:
This functionality here and below must be added to corresponding handler

Reply:
No, your functionality must be accessible thru network. So corresponding handler must do this work. Check 'core/handler' for examples.